src/main/java/com/app/onlinebookstore/repository/BookRepository.java

```diff
@@ -1,8 +1,11 @@
 package com.app.onlinebookstore.repository;
 
 import com.app.onlinebookstore.model.Book;
+import java.util.List;
+import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
 
 public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificationExecutor<Book> {
+    List<Book> findAllByCategoriesId(Long categoriesId, Pageable pageable);
```

Comment:
Check if u don't have n+1 problem without join fetch

Reply:
Yes, if result of this method uses categories of a book then you'll end up with n+1, so add fetch of entity graph